plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,457 @@
+package nfsclient
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type NFSClient struct {
+	Fullstat          bool
+	IncludeMounts     []string
+	ExcludeMounts     []string
+	IncludeOperations []string
+	ExcludeOperations []string
+	Log               telegraf.Logger
+}
+
+var sampleConfig = `
+  # Read more low-level metrics (optional, defaults to false)
+  fullstat = false
+
+  # List of mounts to explictly include or exclude (optional)
+  # The pattern (Go regexp) is matched against the mount point (not the
+  # device being mounted).  If include_mounts is set, all mounts are ignored
+  # unless present in the list. If a mount is listed in both include_mounts
+  # and exclude_mounts, it is excluded.  Go regexp patterns can be used.
+  include_mounts = []
+  exclude_mounts = []
+
+  # List of operations to include or exclude from collecting.  This applies
+  # only when fullstat=true.  Symantics are similar to {include,exclude}_mounts:
+  # the default is to collect everything; when include_operations is set, only
+  # those OPs are collected; when exclude_operations is set, all are collected
+  # except those listed.  If include and exclude are set, the OP is excluded.
+  # See /proc/self/mountstats for a list of valid operations; note that
+  # NFSv3 and NFSv4 have different lists.  While it is not possible to
+  # have different include/exclude lists for NFSv3/4, unused elements
+  # in the list should be okay.  It is possible to have different lists
+  # for different mountpoints:  use mulitple [[input.nfsclient]] stanzas,
+  # with their own lists.  See "include_mounts" above, and be careful of
+  # duplicate metrics.
+  include_operations = []
+  exclude_operations = []
+`
+
+func (n *NFSClient) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *NFSClient) Description() string {
+	return "Read per-mount NFS metrics from /proc/self/mountstats"
+}
+
+func convertToInt64(line []string) []int64 {
+	var nline []int64
+	for _, l := range line[1:] {
+		val, err := strconv.ParseInt(l, 10, 64)
+		if err != nil {
+			if numError, ok := err.(*strconv.NumError); ok {
+				if numError.Err == strconv.ErrRange {
+					log.Fatalf("ErrRange: %v [%v] -> [%v]\n", line, l, val)
+				}
+			}
+		}
+		nline = append(nline, val)
```

Comment:
One question here: If there is an error and the error is **not** `strconv.NumError` you append `val` despite the error condition. Is this wisely or should we do
```suggestion
		if err != nil {
			if numError, ok := err.(*strconv.NumError); ok {
				if numError.Err == strconv.ErrRange {
					log.Fatalf("ErrRange: %v [%v] -> [%v]\n", line, l, val)
				}
			}
			continue
		}
		nline = append(nline, val)
```
to skip that entry?

Reply:
No, the primary logic of the code relies on this...."fall through" behavior.  

This is a similar problem to what I mentioned in https://github.com/influxdata/telegraf/pull/4615#discussion_r548631396   The core logic requires ugly handling of failed conversions.  If someone wants to rewrite that, I think that everyone (especially me) would welcome code contributions.

Reply:
My point was that you get an error but append the line anyway. If this is your intention, it would be nice to add a small comment in the code as this is not what you usually expect if you see error handling.

Reply:
Yeah, the longer I look at this code, the less I like it.  Ultimately, I blame the format of source data:  /proc/self/mountstats.  It's a wreck to parse cleanly.  Comment added.

Reply:
I agree. Can we please explicitly set a value (e.g. `0`) for `val` in the error case and move this part
```
On an error, check for ErrRange, and throw a fatal error
	   if found.  This situation indicates a pretty major issue in
	   the /proc/self/mountstats file, and returning faulty data
	   is worse than no data.  Other errors are ignored, and append
	   whatever we got in the first place (probably 0).
	   Yes, this is ugly.
```
of the comment there? I also see no clean way of dealing with this... :-(